src/tools/idlc/src/descriptor.c

```diff
@@ -1264,21 +1268,21 @@ emit_declarator(
         return ret;
     }
 
-    if ((ret = emit_array(pstate, revisit, path, node, user_data)))
+    if ((ret = emit_array(pstate, revisit, path, node, user_data)) < 0)
       return ret;
 
     /* in case there is no revisit required (array has simple element type) we have
        to close the mutable member immediately, otherwise close it when revisiting */
     if (mutable_aggr_type_member && (!(ret & IDL_VISIT_REVISIT) || revisit)) {
-      if ((ret = close_mutable_member(pstate, descriptor, ctype)))
-        return ret;
+      ret = close_mutable_member(pstate, descriptor, ctype);
```

Comment:
It would be better to use a local `ret` variable here and not reuse `ret` that has the result of `emit_array`, so that the return value of `emit_array` (which can have instructions for the visit function) is returned from this function.